### PR TITLE
Added ignore_forbidden kwarg to sync_all_application_commands, fixed …

### PR DESCRIPTION
**Pull request by alentoghostflame**
…subcommand function names not being fetched properly from CallbackWrapper

Signed-off-by: Alex Schoenhofen <alexanderschoenhofen@gmail.com>

## Checklist

- [X] If code changes were made then they have been tested.
  - [X] I have updated the documentation to reflect the changes.
  - [ ] I have run `task pyright` and fixed the relevant issues.
- [X] This PR fixes an issue.
- [X] This PR adds something new (e.g. new method or parameters).
- [X] This PR is a breaking change (e.g. methods or parameters removed/renamed) (technically???)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by ooliver1 — CHANGES_REQUESTED** [review]

File: nextcord/client.py [hunk 1]
```diff
@@ -2077,6 +2078,9 @@ async def sync_all_application_commands(
         register_new: :class:`bool`
             If a local command that doesn't have a basic match on Discord should be added to Discord.
             Defaults to `True`
+        ignore_forbidden: :class:`bool`
```
[the remark this answers is not in the document]
Comment (alentoghostflame) [reply nextcord/client.py:2081]: Added.

File: nextcord/state.py [hunk 2]
```diff
@@ -718,12 +723,21 @@ async def sync_all_application_commands(
...
+                            data[guild_id] = await self.http.get_guild_commands(
+                                self.application_id, guild_id
+                            )
+                            _log.debug(
+                                "Fetched guild application command data for guild ID %s", guild_id
+                            )
+                        except Forbidden as e:
+                            if ignore_forbidden:
+                                _log.warning(
+                                    f"nextcord.Client: Forbidden error for {guild_id}, is the commands Oauth scope "
```
[the remark this answers is not in the document]
Comment (alentoghostflame) [reply nextcord/state.py]: Fixed.

**Review by ooliver1 — APPROVED** [review]
tested

File: nextcord/client.py [hunk 3]
```diff
@@ -2063,20 +2064,24 @@ async def sync_all_application_commands(
...
         update_known: :class:`bool`
             If commands on Discord have a basic match with a local command, but don't fully match, should be updated.
-            Defaults to `True`
+            Defaults to ``True``
         register_new: :class:`bool`
             If a local command that doesn't have a basic match on Discord should be added to Discord.
-            Defaults to `True`
+            Defaults to ``True``
+        ignore_forbidden: :class:`bool`
+            If this command should raise an :class:`errors.Forbidden` exception when the bot encounters a guild where
```
Comment (Skelmis) [inline nextcord/client.py:2082]: This doc-string indicates the opposite of the actual behavior.
It indicates to me that if `ignore_forbidden` is `True`, it should error on forbidden guilds. While the actual behavior is that it ignores them

Comment (Skelmis) [reply nextcord/client.py:2082]: You would likely be better suited to:
```
If this command should suppress a :class:`errors.Forbidden` exception when the bot encounters a guild where it doesn't have permissions to view application commands.
```

**Review by Skelmis — CHANGES_REQUESTED** [review]

File: nextcord/client.py [hunk 4]
```diff
@@ -2063,20 +2064,24 @@ async def sync_all_application_commands(
...
+            it doesn't have permissions to view application commands.
+            Defaults to ``True``
```
Comment (ooliver1) [inline nextcord/client.py:2084]:
```suggestion
        ignore_forbidden: :class:`bool`
            If this command should suppress a :class:`errors.Forbidden` exception when the bot encounters a guild
            where it doesn't have permissions to view application commands.
            Defaults to ``True``
```
suggestion block for the change above

**Review by Skelmis — APPROVED** [review]

**Pull request merged**